core/src/main/java/org/elasticsearch/search/aggregations/pipeline/InternalSimpleValue.java

```diff
@@ -85,4 +85,16 @@ public XContentBuilder doXContentBody(XContentBuilder builder, Params params) th
         }
         return builder;
     }
+
+    @Override
+    protected int doHashCode() {
+        return Objects.hash(value, format.getWriteableName());
```

Comment:
I see, this is a bit hard to get though: InternalNumericMetricsAggregation#hashCode() checks format and calls super.hashCode(), which in turn calls 'doHashCode' for the implementing subclasses. I get it now, will leave format out here (I think it might be used in other InternalXY classes as well).